shared/Reusable queries for data extraction/query-cohort-matching-yob-sex-imd-index-date.sql

```diff
@@ -0,0 +1,146 @@
+--┌──────────────────────────────────────────────────────────────────┐
+--│ Cohort matching on year of birth / sex / imd / and an index date │
+--└──────────────────────────────────────────────────────────────────┘
+
+-- OBJECTIVE: To take a primary cohort and find a 1:n matched cohort based on year of birth,
+--            sex, imd, and an index date of an event.
+
+-- INPUT: Takes two parameters
+--  - yob-flex: integer - number of years each way that still allow a year of birth match
+--  - index-date-flex: integer - number of days either side of the index date that we allow matching
+--  - num-matches: integer - number of matches for each patient in the cohort
+-- Requires two temp tables to exist as follows:
+-- #MainCohort (FK_Patient_Link_ID, IndexDate, Sex, YearOfBirth,IMD2019Quintile1IsMostDeprived5IsLeastDeprived)
+-- 	- FK_Patient_Link_ID - unique patient id
+--	- IndexDate - date of event of interest (YYYY-MM-DD)
+--	- Sex - M/F
+--	- YearOfBirth - Integer
+--  - IMD2019Quintile1IsMostDeprived5IsLeastDeprived - Integer 1-5
+-- #PotentialMatches (FK_Patient_Link_ID, IndexDate, Sex, YearOfBirth, IMD2019Quintile1IsMostDeprived5IsLeastDeprived)
+-- 	- FK_Patient_Link_ID - unique patient id
+--	- IndexDate - date of event of interest (YYYY-MM-DD)
+--	- Sex - M/F
+--	- YearOfBirth - Integer
+--  - IMD2019Quintile1IsMostDeprived5IsLeastDeprived - Integer 1-5
+
+-- OUTPUT: A temp table as follows:
+-- #CohortStore (FK_Patient_Link_ID, YearOfBirth, Sex, IMD2019Quintile1IsMostDeprived5IsLeastDeprived, IndexDate, MatchingPatientId, MatchingYearOfBirth, MatchingIndexDate)
+--  - FK_Patient_Link_ID - unique patient id for primary cohort patient
+--  - YearOfBirth - of the primary cohort patient
+--  - Sex - of the primary cohort patient
+--  - IMD2019Quintile1IsMostDeprived5IsLeastDeprived - of the primary cohort patient
+--  - IndexDate - date of event of interest (YYYY-MM-DD)
+--  - MatchingPatientId - id of the matched patient
+--  - MatchingYearOfBirth - year of birth of the matched patient
+--  - MatchingIndexDate - index date for the matched patient
+
+-- First we extend the #PrimaryCohort table to give each age-sex combo a unique number
```

Comment:
should this comment be 'each age-sex-IMDQuintile-IndexDate' combo?

Reply:
Yes good point. I've updated here and in a few other places in the file.